packages/core/src/initializer/index.ts

```diff
@@ -16,13 +16,34 @@ import { StrykerInquirer } from './stryker-inquirer.js';
 import { createInitializers } from './custom-initializers/index.js';
 import { GitignoreWriter } from './gitignore-writer.js';
 
-const NPM_REGISTRY = 'https://registry.npmjs.com';
+let currentRegistry: string | undefined;
+
+function getRegistry(): string {
+  if (!currentRegistry) {
+    if (process.env.npm_config_registry) {
+      currentRegistry = process.env.npm_config_registry;
+    } else if (process.env.npm_command) {
+      // if running inside npm and not having the registry than it's the default one
+      currentRegistry = 'https://registry.npmjs.com';
+    } else {
+      // Using global as when trying to get the registry inside npm workspace it would fail
+      const registry = execaSync('npm', ['config', 'get', '--global', 'registry'], {
+        stdout: 'pipe',
+        timeout: 20000,
+      });
+
+      currentRegistry = registry.stdout;
+    }
+  }
+
+  return currentRegistry;
+}
 
 export function initializerFactory(): StrykerInitializer {
   LogConfigurator.configureMainProcess(LogLevel.Information);
   return provideLogger(createInjector())
     .provideValue(initializerTokens.out, console.log)
-    .provideValue(initializerTokens.restClientNpm, new RestClient('npm', NPM_REGISTRY))
```

Comment:
Could you instead use the dependency injection mechanism to provide a cached version of the npm registry? That way we get caching for free and it will be far easier to test.

Pseudo code:

```js
function createNpmRegistryClient(npmRegistry: string) {
  return new RestClient('npm', npmRegistry);
}
createNpmRegistryClient.inject = [initializerTokens.npmRegistry] as const;

// inside the chain
.provideFactory(initializerTokens.npmRegistry, getRegistry)
.provideFactory(initializerTokens.restClientNpm, createRestClient)
```